### PR TITLE
[mongodb] Update the pom.xml to make checkstyle fail the build. Fixes #365

### LINKED ISSUE #365: Make checkstyle enforcing for the MongoDB client.
I have been getting annoyed with the multiple formats rolling around so I have started with my own yard and made the checkstyle enforcing for the MongoDB client.

To get the build to pass most of the changes were done via automated formatters/cleanup. The rest where variable name changes.

The bad part of doing this kind of thing is it creates a patch horizon that it is extremely difficult to move patches across (in either direction). I think the consistency is worth the one time pain.

### PR DESCRIPTION
**Pull request by allanbank**
Update the pom.xml to make checkstyle fail the build. 
Updated code to satisfy the checkstyle validation.

Fixes #365

### CODE REVIEW

File: mongodb/src/main/java/com/yahoo/ycsb/db/MongoDbClient.java [hunk 1]
```diff
@@ -1,12 +1,19 @@
-/**
- * MongoDB client binding for YCSB.
+/*
+ * Copyright (c) 2014, Yahoo!, Inc. All rights reserved.
```
Comment (busbey) [inline mongodb/src/main/java/com/yahoo/ycsb/db/MongoDbClient.java]: We don't have any kind of CLA that assigns copyrights to Yahoo!, so let's leave out adding these for now. I've been trying to think of the best way to handle #313 and all I've got so far is that adding Yahoo copyright statements isn't the way to go. :)

If they're already present, please leave them. That's a whole different mess.

File: mongodb/src/main/java/com/yahoo/ycsb/db/package-info.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+/**
+ * The YCSB driver for <a href="https://www.mongodb.org/">MongoDB</a>.
+ * Supports both the
+ * <a href="http://www.allanbank.com/mongodb-async-driver/">Asynchronous
+ * Java Driver</a> and the MongoDB Inc.
+ * <a href="http://docs.mongodb.org/ecosystem/drivers/java/">driver</a>.
```
Comment (busbey) [inline mongodb/src/main/java/com/yahoo/ycsb/db/package-info.java]: nit: maybe a reference to check the README?

**Comment by busbey** [comment]
with the copyright statement thing above addressed, I'm +1.

**Comment by busbey** [comment]
(for other reviewiers, [the fill diff without whitespace changes](https://github.com/brianfrankcooper/YCSB/pull/366/files?w=1) helped speed things up.)

**Comment by cmatser** [comment]
Should there be a "checkstyle.xml" file included in the PR?
 <configLocation>checkstyle.xml</configLocation>

**Comment by allanbank** [comment]
@cmaster  - Interesting. Checkstyle was checking based on a config somewhere...  I'll be honest I and say I just copied the blob from the root pom and then flipped the enforcing flags on. There is the checkstyle.xml file in the root directory and I just assumed it was using that one.  I will see if setting the path to ```../checkstyle.xml``` makes a difference.

Maybe a better approach to this is to make the root pom enforcing and then update all of the poms except the MongoDB pom to override the configuration to not be enforcing. Then as we knock out sub-projects we just remove checkstyle item from the poms.

**Comment by allanbank** [comment]
FYI - I did a force push of the amended commit to remove the bad copyright header.

I also changed the path to the checkstyle.xml to be relative and it had not effect. I left it that way in the commit as I think it is clearer that it is referencing the file one directory up.

**Comment by cmatser** [comment]
Nice work!

**Pull request merged**